pkg/controller/install/certresources.go

```diff
@@ -260,9 +266,8 @@ func (i *StrategyDeploymentInstaller) installCertRequirementsForDeployment(deplo
 		service.SetOwnerReferences(existingService.GetOwnerReferences())
 
 		// Delete the Service to replace
-		deleteErr := i.strategyClient.GetOpClient().DeleteService(service.GetNamespace(), service.GetName(), &metav1.DeleteOptions{})
-		if deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
-			return nil, nil, fmt.Errorf("could not delete existing service %s", service.GetName())
+		if err := i.deleteServiceWithTimeout(service, serviceDeletionTimeout); err != nil {
+			return nil, nil, err
```

Comment:
I think we briefly talked about this yesterday, but did you end up finding any context for why we're creating/deletion these resources vs. creating/updating/etc.? Was it a technical limitation at the time? Could we use SSA in our controllers?

Reply:
@timflannagan I didn't manage to get to the bottom of it =S @awgreene @kevinrizza do you guys happen to know?

Reply:
I believe its because a number of these resources don't support update commands on certain fields, IIRC it originally surfaced with a service update request?